pkg/syndesis/action/upgrade.go

```diff
@@ -149,7 +149,7 @@ func (a *Upgrade) Execute(syndesis *v1alpha1.Syndesis) error {
 
 func upgradeCompleted(syndesis *v1alpha1.Syndesis, newVersion string) error {
 	target := syndesis.DeepCopy()
-	target.Status.InstallationStatus = v1alpha1.SyndesisInstallationStatusInstalled
+	target.Status.InstallationStatus = v1alpha1.SyndesisInstallationStatusAttaching
```

Comment:
tbh I would do the owner attachment directly after an successful upgrade and during the creation of the synthetic Syndesis resource for legacy installtation.